Warning: Use of undefined constant STDIN - assumed 'STDIN'

Raxmatilla97 opened the pull request:
Warning: Use of undefined constant STDIN - assumed 'STDIN' (this will throw an Error in a future version of PHP) in C:\Users\Professional\Downloads\OpenServer\domains\vufind-dev\vendor\symfony\console\Helper\QuestionHelper.php on line 108

Warning: fgets() expects parameter 1 to be resource, string given in C:\Users\Professional\Downloads\OpenServer\domains\vufind-dev\vendor\symfony\console\Helper\QuestionHelper.php on line 130

Commit:
Add support for tagging of user lists.
- Deprecates VuFind\Db\Row\UserList::getTags() in favor of VuFind\Db\Row\UserList::getResourceTags()
- Deprecates VuFind\Db\Table\ResourceTags::destroyLinks() in favor of VuFind\Db\Table\ResourceTags::destroyResourceLinks()
- Includes code originally developed as part of pull request #1645.

demiankatz committed:
Merge branch 'release-7.0' into dev

bbusenius committed:
FOLIO: use location discoveryDisplayName instead of name (#1692)

Commit:
Fix deleting favorite from userlist. (#1693)

- Fixes a bug introduced in 2b05cef5c7f3041444ad415935f482e8b3cacfee

demiankatz committed:
Merge branch 'dev'

demiankatz committed:
Use SQL_STAR constant for clarity. (#1690)

crhallberg committed:
Create CODE_OF_CONDUCT.md (#1685)

crhallberg committed:
Create CONTRIBUTING.md (#1686)

demiankatz committed:
Merge branch 'release-7.0' into dev

Commit:
Add option to populate ListItems channel using list tags (#1645)

hajoseng committed:
Restrict facet values by prefixes and/or matches on a per field basis. (#1694)

- This is done by setting the arrays facet_prefix_by_field or facet_matches_by_field within the Results_Settings part of facets.ini.

demiankatz committed:
Merge branch 'release-7.0' into dev

demiankatz committed:
Merge branch 'release-7.0' into dev

demiankatz committed:
Fix bug: $replyToName was never used.
- Resolves VUFIND-1419.

demiankatz committed:
Merge branch 'release-7.0' into dev

demiankatz committed:
Upgrade YAML library. (#1696)

demiankatz committed:
Add mink test for list tags (#1697)

- Also includes minor style improvement to tag display
- Also fixes order of existing assertions in FavoritesTest
- Resolves VUFIND-1416.

demiankatz committed:
Merge branch 'release-7.0' into dev

EreMaijala committed:
KohaRest: Add a comment about OAuth2 authorization header. (#1700)

- Also makes the comment block a bit more readable.

demiankatz committed:
Merge branch 'release-7.0' into dev

demiankatz committed:
Merge branch 'release-7.0' into dev

demiankatz committed:
Minor language tweaks.

demiankatz committed:
Merge branch 'release-7.0' into dev

demiankatz committed:
Merge branch 'release-7.0' into dev

demiankatz committed:
Merge branch 'release-7.0' into dev

demiankatz committed:
Merge branch 'release-7.0' into dev

demiankatz committed:
Add permission support to combined search (#1707)

- Resolves VUFIND-1421.

demiankatz committed:
Merge remote-tracking branch 'origin/release-7.0' into dev

demiankatz committed:
Upgrade to League/CommonMark 1.5 (#1699)

xmorave2 committed:
Significant improvements to XCNCIP2 ILS driver (#1672)

- Adds tests and fixes issues revealed by testing
- Improves code style
- Adds new functionality, including configuration settings for "fromAgency" and optional location loading via "Lookup Agency" service.

domerz committed:
Optionally allow rendering of a custom template for unavailable covers (#1679)

demiankatz committed:
Merge branch 'release-7.0' into dev

demiankatz committed:
Merge branch 'release-7.0' into dev

bbusenius committed:
FOLIO: Modify notes field and add item_notes, issues, supplements and indexes (#1708)

demiankatz committed:
Merge branch 'release-7.0' into dev

xmorave2 committed:
Item id is not required by NCIP, could be null

xmorave2 committed:
Add configuration for new abilities of CommonMark 1.5 (#1716)

- Update League/CommonMark to version 1.5.5
- Add new extensions to markdown.ini
- Update MarkdownFactory

demiankatz committed:
Merge branch 'release-7.0' into dev

xmorave2 committed:
Update ItemElementType schema to that one accepted by xcnciptoolkit

demiankatz committed:
Refactor code to make username easier to override.

demiankatz committed:
Merge branch 'release-7.0' into dev

EreMaijala committed:
Mechanism to specify extra scripts for tabs in RecordTabs.ini.
- This is not used yet, but will be required for PR #1581.

demiankatz committed:
Merge branch 'release-7.0' into dev

demiankatz committed:
Reduce EDS API calls by caching data required for Search Options. (#1688)

EreMaijala committed:
SierraRest: Refactor authorization grant requesting (#1658)

- Now more robust; works with curl adapter and doesn't rely on the number of redirects.

EreMaijala committed:
[VUFIND-1422] Fix a couple of prefix handling issues in the MultiBackend driver. (#1711)

demiankatz committed:
Merge branch 'release-7.0' into dev

EreMaijala committed:
Alma: Clean up  error messages and always use logger. (#1723)

xmorave2 committed:
Fix double colon in MARC links (#1725)

crhallberg committed:
Lightbox Keyboard Support / Improved Accessibility (#1667)

- Constrain keyboard focus to lightbox. Add ESC to close lightbox.
- Adjust lightbox close button to look better and be more accessible.

xmorave2 committed:
Add ability to disable renewals to XCNCIP2 driver

demiankatz committed:
Merge branch 'release-7.0' into dev

crhallberg committed:
Take a screenshot on integrated test failure. (#1562)

- Includes some related fixes to AutoRetryTrait.

Commit:
FeedbackForms: allow extending Form class. (#1724)

EreMaijala committed:
Create FieldSpecTools SolrMarc class.
- Refactor useful utility method to shared location.
- Add additional utility method (not used yet, but required by #1581).

demiankatz committed:
Merge branch 'release-7.0' into dev

EreMaijala committed:
SierraRest: Fix a couple of holdings retrieval issues (#1726)

- Avoid checking holdings for status requests.
- Accept also fixed field with key 40 as holdings location. (This method should work also if the field doesn't have a label. I didn't dare to remove the label check, though.)

EreMaijala committed:
Merge branch 'release-7.0' into dev

Wdcus committed:
Update Evergreen.php getStatus() and getHoldings() to exclude deleted items (#1728)

Co-authored-by: Wes <wesley.custer@gmail.com>

demiankatz committed:
Merge branch 'release-7.0' into dev

EreMaijala committed:
Add requirements to ImageFactoryTest. (#1731)

xmorave2 committed:
Fix markdown factory not getting config from markdown.ini (#1734)

EreMaijala committed:
Handle null response from Solr properly. (#1733)

At least the similar handler in Solr 8.x can return null in the response element when no similar records are found.

EreMaijala committed:
Annotate getParsedTagList with @Deprecated.

demiankatz committed:
Merge branch 'release-7.0' into dev

EreMaijala committed:
Reset authentication state on logout. (#1735)

- This probably has no effect in real life unless the followup after logout is short-circuited.

EreMaijala committed:
Make it possible to use smaller expiration than 2 days for sessions. (#1736)

- Allows defining a smaller minimum age than default for expire commands. Also makes sure that fractional values are printed with '.' as the decimal separator.

EreMaijala committed:
Catch ILS exception from patronLogin properly when modifying a library card. (#1738)

EreMaijala committed:
SierraRest driver: exception handling improvements (#1737)

- Catch HTTP client's exceptions in SierraRest driver (otherwise e.g. curl adapter's timeout exception goes unhandled).
- Be sure to include previous exception in ILS exceptions.

demiankatz committed:
Merge remote-tracking branch 'origin/release-7.0' into dev

demiankatz committed:
Merge branch 'release-7.0' into dev

EreMaijala committed:
Make sure getFilterStringParts always returns TranslatableString objects. (#1740)

xmorave2 committed:
Refactor creating http client in ObalkyKnih service (#1741)

EreMaijala committed:
Remove checkRoute=1 parameter from the url when it's no longer needed. (#1742)

EreMaijala committed:
Fix assertEquals parameter order in HierarchicalFacetHelperTest.

EreMaijala committed:
Fix undefined offset when creating hierarchical facet list.

This could have been encountered in advanced search form.

demiankatz committed:
Merge branch 'release-7.0' into dev

EreMaijala committed:
Add pagination support to getUserTransactions (#1739)

- This allows the call to return proper data when checkouts are paged, even if slowly.

EreMaijala committed:
KohaRest: Allow mapping of different NotForLoan statuses separately. (#1730)

demiankatz committed:
Merge branch 'release-7.0' into dev

xmorave2 committed:
ObalkyKnihService: catch exception on HTTP request (#1746)

demiankatz committed:
Merge branch 'release-7.0' into dev

demiankatz commented:
@Raxmatilla97, this pull request does not appear to be formatted correctly -- it is attempting to merge the 'dev' branch into the 'release-7.0' branch. Is there a branch you created somewhere else in a fork which addresses the problem you have cited?

If you are simply trying to report a problem, please use [JIRA](https://vufind.org/jira) instead; that is the project's issue tracker.

The error you describe sounds like it may be the consequence of trying to run Console-oriented code in the web interface, but I'm not sure how/why that would be happening. Steps to reproduce the issue would be helpful. Please share them on the JIRA if possible.

In any case, thanks for reporting the issue, and please let me know if you have questions or if I can do anything more to help!

hajoseng committed:
Configurable record retrieval batch size for Solr (#1749)

- Making the limit of records to retrieve per query in a batch query configurable

demiankatz committed:
Use more appropriate currency icon. (#1750)

powerriegel committed:
Better implementation of getPlayingTimes in MarcAdvancedTrait (#1751)

ihardy committed:
return null if lc call number is empty (#1752)

rajaro committed:
improve record tab hash handling (#1754)

demiankatz committed:
Merge branch 'release-7.0' into dev

demiankatz committed:
Merge branch 'release-7.0' into dev

demiankatz committed:
Merge branch 'release-7.0' into dev

xmorave2 committed:
Make the max limit of search API results more flexible (#1762)

EreMaijala committed:
KohaRest: Split GetItemStatusCodes into smaller sub-methods. (#1756)

EreMaijala committed:
Treat invalid date errors from Solr as parser errors. (#1763)

demiankatz committed:
Use mixed instead of any in doc comments.

demiankatz committed:
Add missing use statement.

demiankatz committed:
Begin fixing return types using Psalm. (#1765)

demiankatz committed:
Simplify PHPUnit fixture handling. (#1766)

- Create trait to centralize fixture loading
- Eliminate unnecessary constants used by test process
- Reduce usage of hard-to-read relative paths

demiankatz committed:
Merge branch 'release-7.0' into dev

EreMaijala committed:
KohaRest: Add support for mapping different "Lost" statuses. (#1771)

demiankatz committed:
Use GET instead of POST for limit control. (#1727)

- Creates cleaner browser history (no back button problems, etc.).

EreMaijala committed:
Re-set keyboard focus in lightbox after the content has loaded. (#1767)

demiankatz committed:
Merge branch 'release-7.0' into dev

demiankatz committed:
Merge branch 'release-7.0' into dev

demiankatz committed:
Merge branch 'release-7.0' into dev

crhallberg committed:
Return to previous page after printing. (#1753)

demiankatz committed:
Merge branch 'release-7.0' into dev

demiankatz committed:
Improve MARC table of contents processing. (#1770)

EreMaijala committed:
Add tests that cover most of MarcBasicTrait and MarcAdvancedTrait. (#1777)

demiankatz committed:
Merge branch 'release-7.0' into dev

demiankatz committed:
Log writer for Office365 webhooks. (#1757)

xmorave2 committed:
Change default content type of ajax responses to application/json (#1780)

- According to https://www.ietf.org/rfc/rfc4627.txt

demiankatz committed:
Add test for DOI recommendation module.

demiankatz committed:
Add config setting to control DOI auto-redirect (#1781)

demiankatz committed:
Implement getContainerEndPage() for EDS (#1784)

demiankatz committed:
Index DOIs from 856 fields. (#1783)

demiankatz committed:
Add more record driver tests and improve code (#1785)

- Add tests for SolrMarcRemote and SolrOverdrive
- Includes general style cleanup/improvement
- Fixes a bug (typo) in SolrOverdrive
- Removes some unneeded SolrOverdrive methods (they were trying to work around MARC-specific logic that doesn't actually exist in SolrMarc)

demiankatz committed:
Update composer dependencies.

demiankatz committed:
More error-tolerant tests.
- If jQuery isn't ready, wait for it.

demiankatz committed:
More appropriate login link for EDS.

demiankatz committed:
EDS improvement: retrieve subtitles correctly.

xrosecky committed:
Disable deduplication listener when search parameters contains child filter (#1789)

demiankatz committed:
Style fix.

demiankatz committed:
Fix inaccurate @return annotations.

demiankatz committed:
Add missing use statements to factories (#1790)

demiankatz committed:
Fix typo.

demiankatz committed:
Merge branch 'release-7.0' into dev

demiankatz committed:
Merge branch 'release-7.0' into dev

demiankatz committed:
Merge branch 'release-7.0' into dev

demiankatz committed:
Merge branch 'release-7.0' into dev

demiankatz committed:
Merge branch 'release-7.0' into dev

demiankatz committed:
Updated translations courtesy of Dina Hashim/Knowledgeware Technologies.

demiankatz committed:
Brazilian Portuguese translations courtesy of Diego Macêdo.

demiankatz committed:
Updated translations courtesy of Markus Fischer.

demiankatz committed:
Merge branch 'release-7.0' into dev

demiankatz committed:
Simplify language.

xmorave2 committed:
Add Czech translation for 7.1 (#1801)

milotype committed:
Updated Croatian translations.

demiankatz committed:
Update/expand comment.

demiankatz committed:
Updated translations courtesy of Christian Dabrowski.

EreMaijala committed:
Refactor resultsAction() to better support alternate search actions.

EreMaijala committed:
Fix combined search QR code handling.

EreMaijala committed:
Scroll to record tabs on hash change.

demiankatz committed:
Standardize callable typehints.

demiankatz committed:
Merge branch 'release-7.0' into dev

demiankatz committed:
Updated translations courtesy of Mạnh Phan Văn.

demiankatz committed:
Merge branch 'release-7.0' into dev

demiankatz committed:
Updated translations courtesy of Keiji Suzuki.

demiankatz committed:
Merge branch 'release-7.0' into dev

demiankatz committed:
Updated translations courtesy of Hugo Agud.

demiankatz committed:
Updated translations courtesy of Theodoros Theodoropoulos.

demiankatz committed:
Updated translations courtesy of Uğur Bulgan.

EreMaijala committed:
KohaRest: Check that item_id is set before sending it to Koha. (#1809)

- It might not be if it's missing from HMACKeys.

EreMaijala committed:
Add possibility to reset Demo driver's cache.

This can be accomplished by adding clear_demo=1 parameter to any VuFind url that uses the Demo driver.

demiankatz committed:
Updated translations courtesy of Agnès Manneheut.

demiankatz committed:
Updated translations courtesy of Jorge Ramírez.

Commit:
FeedbackForms: require one checkbox option to be selected (#1811)

demiankatz committed:
Updated translations courtesy of Anneke Goovaerts.

EreMaijala committed:
Improve some Finnish and Swedish translations for course reserves. (#1816)

mtrojan-ub committed:
Refer to $checkboxFilters and $filterList via $this (#1818)

Commit:
FeedbackForm: improve validation of checkbox options. (#1814)

bbusenius committed:
Configurable top facet override of alphabetical sort for limits (#1807)

xrosecky committed:
Shibboleth improvements (#1617)

- Made session expiration check behavior configurable
- Added support / config setting for reading Shibboleth attributes via headers
- Added support for multiple IdPs (useful in a MultiBackend setup)
- Significant code refactoring/improvement.

demiankatz committed:
Upgrade to Solr 7.7.3. (#1793)

EreMaijala committed:
Explicitly initialize OpenURL embedding in ajax-loaded record tabs. (#1819)

EreMaijala committed:
Add transEscAttr view helper and use it in templates. (#1820)

olli-gold committed:
makes MARC record field configurable (#1813)

EreMaijala committed:
Add support for clearing account cache based on status information. (#1804)

- The cache is automatically refreshed if the information on any of the actual MyResearch pages doesn't match the cached status.

demiankatz committed:
Adjust timeouts in tool configurations. (#1822)

- Reduces the chance of problems caused by slow Solr download during "composer install"

EreMaijala committed:
[VUFIND-120] Add support for record "versions" display. (#1581)

demiankatz commented:
Because this appears to have been opened in error, and because there has been no further discussion since my inquiry in October, I am closing this PR on the assumption that it is not needed. If I can help with anything, please let me know, @Raxmatilla97!
